src/services/item/plugins/geolocation/repository.test.ts

```diff
@@ -0,0 +1,231 @@
+      expect(res).toBeNull();
+    });
+  });
+
+  describe('getItemsIn', () => {
+    it('returns items in box', async () => {
+      const { item: parentItem } = await saveItemAndMembership({ member: actor });
+      const { item } = await saveItemAndMembership({ member: actor, parentItem });
+      const geoloc = { lat: 1, lng: 2, item, country: 'de' };
+      await rawRepository.save(geoloc);
+      const geolocParent = { lat: 1, lng: 2, item: parentItem, country: 'de' };
+      await rawRepository.save(geolocParent);
+
+      // noise
+      await saveItemAndMembership({ member: actor, parentItem });
+
+      const res = await repository.getItemsIn(0, 4, 0, 4);
+      expect(res).toHaveLength(2);
+      expect(res).toContainEqual(geoloc);
+      expect(res).toContainEqual(geolocParent);
+    });
+
+    it('returns nothing', async () => {
+      const { item: parentItem } = await saveItemAndMembership({ member: actor });
+      const { item } = await saveItemAndMembership({ member: actor, parentItem });
+      const geoloc = { lat: 1, lng: 2, item, country: 'de' };
+      await rawRepository.save(geoloc);
+      const geolocParent = { lat: 1, lng: 2, item: parentItem, country: 'de' };
+      await rawRepository.save(geolocParent);
+
+      // noise
+      await saveItemAndMembership({ member: actor, parentItem });
+
+      const res = await repository.getItemsIn(0, 0.5, 0, 0.5);
+      expect(res).toHaveLength(0);
+    });
+
+    it('returns with swapped values', async () => {
+      const { item: parentItem } = await saveItemAndMembership({ member: actor });
+      const { item } = await saveItemAndMembership({ member: actor, parentItem });
+      const geoloc = { lat: 1, lng: 2, item, country: 'de' };
+      await rawRepository.save(geoloc);
+      const geolocParent = { lat: 1, lng: 2, item: parentItem, country: 'de' };
+      await rawRepository.save(geolocParent);
+
+      // noise
+      await saveItemAndMembership({ member: actor, parentItem });
+
+      const res = await repository.getItemsIn(4, 0, 4, 0);
+      expect(res).toHaveLength(2);
+      expect(res).toContainEqual(geoloc);
+      expect(res).toContainEqual(geolocParent);
+    });
+  });
+
+  describe('put', () => {
+    it('create new geolocation for item', async () => {
+      const { item } = await saveItemAndMembership({ member: actor });
+
+      const lat = 40.785091;
+      const lng = -73.968285;
+      await repository.put(item.path, lat, lng);
+      const geoloc = await rawRepository.findOneBy({ lat, lng });
+      expect(geoloc).toMatchObject({ lat, lng, country: 'AQ' });
+    });
+    it('create new geolocation that does not have a country', async () => {
+      const { item } = await saveItemAndMembership({ member: actor });
+
+      const lat = 1;
+      const lng = 2;
+      await repository.put(item.path, lat, lng);
+      const geoloc = await rawRepository.findOneBy({ lat, lng });
+      expect(geoloc).toMatchObject({ lat, lng, country: null });
```

Comment:
I'm not sure to understand the difference between this test, and the previous one. How it is possible that this test return a null country ?

Is there any reason to not enforce to always receive a country when saving a lat and long ? I mean, saving the country in the database when inserting or updating lat and long ?

Reply:
I'm accepting points that are in the ocean I guess? but we could enforce points on earth?

Reply:
Yeah I didn't thought about that. I think if it is acceptable to have points in the ocean or something like that, it is not necessary to enforce points on earth ?